File: algorithms/1523,count-odd-numbers-in-an-interval-range/solution.rs
impl Solution {
    pub fn count_odds(low: i32, high: i32) -> i32 {
        let (mut low, mut high) = (low, high);
        let mut sum = (high - low) / 2;

        if (high - low) % 2 == 1 {
            sum += 1;
        } else if low % 2 == 1 {
            sum += 1;
        }

        sum
    }
}

File: algorithms/1523,count-odd-numbers-in-an-interval-range/solution.rb
# @param {Integer} low
# @param {Integer} high
# @return {Integer}
def count_odds(low, high)
  diff = high - low
  sum = diff / 2

  if diff % 2 == 1
    sum += 1
  elsif low % 2 == 1
    sum += 1
  end

  sum
end
